UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs

```diff
@@ -197,5 +189,40 @@ public static ObservationProto ToProto(this Observation obs)
             obsProto.Shape.AddRange(obs.Shape);
             return obsProto;
         }
+
+        public static ObservationProto GetObservationProto(this ISensor sensor, WriteAdapter writeAdapter)
+        {
+            var shape = sensor.GetObservationShape();
+            ObservationProto observationProto = null;
+            if (sensor.GetCompressionType() == SensorCompressionType.None)
+            {
+                var numFloats = sensor.ObservationSize();
+                var floatDataProto = new ObservationProto.Types.FloatData();
+                // Resize the float array
```

Comment:
Unfortunately, doesn't seem to be a way to set the Capacity on this.

Reply:
Actually, there is, but only in newer versions of the library: https://github.com/protocolbuffers/protobuf/commit/da5740082ef7c39a5503e4d57ee73e11d3b43586

Reply:
Added code comment on this too.